### PR TITLE
add submissions model

### LINKED ISSUE #50: Add submission details after submitting any problem
The submission schema should look like:
username: String,
    qID: Number,
    subID: Number,
    code: String,
    language: String,
    verdict: String,
    time: Number,
    memory: Number,
    isVisible: Boolean,
    timeStamp: Date,
    tc: [{ status: String, time: Number, memory: Number }]

The subID will be 1 + total no of submissions till now. URL: http://localhost:3000/submit/:qID

### PR DESCRIPTION
**Pull request by vsvipul**
# What's this PR do? (related issue if exists)
fixes #50

### CODE REVIEW

**Comment by vsvipul** [comment]
isn't number() data type used to store decimal values as well ? https://developer.mozilla.org/en-US/docs/Glossary/Number . Also, need help with why subID is always stored as 1 .

**Comment by Vishal1541** [comment]
In mongoose it doesn't work. It stores integer only.

**Comment by vsvipul** [comment]
@Vishal1541  i have made the required changes.. please review

File: models/submission.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    qID: Number,
+    subID: Number,
+    code: String,
+    language: String,
+    verdict: String,
+    time: Double,
+    memory: Number,
+    isVisible: Boolean,
+    timeStamp: Date,
+    tc: [{ status: String, time: mongoose.Decimal128, memory: Number }]
```
Comment (Vishal1541) [inline models/submission.js]: You can change `time: mongoose.Decimal128` to `time: Double`. It's giving some error on running the mongodb.

File: controls/problems.js [hunk 2]
```diff
@@ -20,7 +60,49 @@ helper.submitSolution = async (req, res, next) => {
...
+                langName = lang[i].name;
+                break; 
+            }
+        }
+        var newSubmission = new submission({
+            username: req.user ? req.user.username : 'Guest',
+            qID: req.body.qID,
+            subID: 1 + subCount,
+            code: req.body.code,
+            language: langName,
```
Comment (Vishal1541) [inline controls/problems.js]: `langName` is not being inserted into the database. Also, there are fewer compilers available now.
One easy thing you could do was either make `lang.json` with all the compilers or let all the compilers be in .ejs itself, but add `data.langCompiler = $("#language option:selected").text();` to `./public/js/solution_submit.js` and directly use `req.body.langCompiler` in `./controls/problems.js`

Comment (vsvipul) [reply controls/problems.js]: please review again @Vishal1541

**Pull request merged**